lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -178,30 +173,43 @@ bool ParseXML::validate_xml(
 bool ParseXML::save_xml(
         xercesc::DOMDocument*& doc)
 {
+    // Config would configure serialized XML data
+    config = serializer->getDomConfig();
+    config->setParameter(xercesc::XMLUni::fgDOMWRTFormatPrettyPrint, true);
+    config->setParameter(xercesc::XMLUni::fgDOMXMLDeclaration, true);
+
     // Save XML document in target file path
     target = new xercesc::LocalFileFormatTarget(xercesc::XMLString::transcode(xml_file.c_str()));
     output->setByteStream(target);
     return serializer->write(doc, output);
 }
 
 xercesc::DOMNode* ParseXML::get_node(
-        xercesc::DOMNodeList*& node_tag_list,
+        xercesc::DOMNode*& parent_node,
         const std::string& tag_name,
-        int32_t index,
+        const std::string* index,
```

Comment:
The method `get_node` was updated with this string pointer as index due to the 5th case consideration:

1. Positive index: return element associated to positive index value. String as "5"
2. Negative index: return element associated to negative index value. String as "-2"
3. Invalid index (unbounded index, or unable to parse it to integer): return BadParameter exception. String as "-2048" or "thisshouldbeaninteger"
4. Empty index: return complete list element (parent). String as ""
5. _Nullptr index: getter should not look for listed elements (should be complex or map elements). String as nullptr._

**This would be modified in future PR to improve the library infraestructure.**